### DISCUSSION TITLE
Dialog role="alertdialog" attribute

### DISCUSSION THREAD

**Discussion by aeirola** [tailwindlabs/headlessui#617] [Ideas]
The react [Dialog](https://headlessui.dev/react/dialog) component is useful for implementing different types of dialogs. There are of course many different types of dialogs, but WAI ARIA defines two main types:

- Dialog: https://www.w3.org/TR/wai-aria-1.1/#dialog
- Alert dialog: https://www.w3.org/TR/wai-aria-1.1/#alertdialog

At the moment though, the Dialog component forces the `role` attribute to be `dialog`. This means that it cannot be used to implement a WAI ARIA `alertdialog`  type of dialog. Wondering if it would be possible to make the `role` attribute configurable in some way through props.

**Comment by reinink** [comment]
Hey! Just letting you know that this has been added: https://github.com/tailwindlabs/headlessui/pull/2709
